homeassistant/components/media_player/samsungtv.py

```diff
@@ -0,0 +1,175 @@
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.samsungtv/
+"""
+import logging
+import socket
+from samsungctl import Remote
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_VOLUME_STEP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_NEXT_TRACK, SUPPORT_TURN_OFF,
+    DOMAIN)
+from homeassistant.const import (
+    CONF_HOST, STATE_OFF,
+    STATE_ON, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_SAMSUNGTV = SUPPORT_PAUSE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_NEXT_TRACK | SUPPORT_TURN_OFF
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Denon platform. """
+    if not config.get(CONF_HOST):
+        _LOGGER.error(
+            "Missing required configuration items in %s: %s",
+            DOMAIN,
+            CONF_HOST)
+        return False
+    name = config.get("name", 'Samsung TV Remote')
+    # Generate a config for the Samsung lib
+    remote_config = {
+        "name": "HomeAssistant",
+        "description": config.get("name", ''),
+        "id": "ha.component.samsung",
+        "port": config.get("port", 55000),
+        "host": config.get("host"),
+        "timeout": config.get("timeout", 0),
+    }
+
+    add_devices([SamsungTVDevice(name, remote_config)])
+
+
+# pylint: disable=too-many-public-methods
+class SamsungTVDevice(MediaPlayerDevice):
+    """ Represents a Denon device. """
+
+    def set_volume_level(self, volume):
+        pass
+
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, name, config):
+        self._name = name
+        # Assume that the TV is not muted
+        self._muted = False
+        # Assume that the TV is in Play mode
+        self._playing = True
+        self._state = STATE_UNKNOWN
+        self._remote = None
+        self._config = config
+
+    def update(self):
+        # Send an empty key to see if we are still connected
+        return self.send_key('KEY_POWER')
+
+    def get_remote(self):
+        """ Creates or Returns a remote control instance """
+        if self._remote is None:
+            # We need to create a new instance to reconnect.
+            self._remote = Remote(self._config)
+
+        return self._remote
+
+    def send_key(self, key):
+        """ Sends a key to the tv and handles exceptions """
+        try:
+            self.get_remote().control(key)
+            self._state = STATE_ON
+        except (Remote.UnhandledResponse, Remote.AccessDenied,
+                BrokenPipeError):
+            # We got a response so it's on.
+            # BrokenPipe can occur when the commands is sent to fast
+            self._state = STATE_ON
+            self._remote = None
+            return False
+        except (Remote.ConnectionClosed, Remote.ConnectionClosed,
```

Comment:
There seems to be no difference between the results of these exceptions and the previous ones. They can be grouped?

Is there a 'main' Exception that all these inherit from that you would be able to catch?